Sources/NIOPosix/System.swift

```diff
@@ -37,7 +37,7 @@ import CNIOWindows
 
 internal typealias MMsgHdr = CNIOWindows_mmsghdr
 #else
-let badOS = { fatalError("unsupported OS") }()
```

Comment:
I have no idea why @weissi used this formulation instead of erroring, as I added now.

Reply:
That was from before Swift had `#error(...)`.